lewis/devices/julabo/device.py

```diff
@@ -0,0 +1,302 @@
+        self.is_circulating = 0  # 0 for off, 1 for on
+        self.temperature_ramp_rate = 5.0  # Guessed value in C/min
+
+        self.internal_p = 0.1  # The proportional
+        self.internal_i = 3  # The integral
+        self.internal_d = 0  # The derivative
+        self.external_p = 0.1  # The proportional
+        self.external_i = 3  # The integral
+        self.external_d = 0  # The derivative
+
+    def _get_state_handlers(self):
+        return {
+            'circulate': states.DefaultCirculatingState(),
+            'not_circulate': states.DefaultNotCirculatingState(),
+        }
+
+    def _get_initial_state(self):
+        return 'not_circulate'
+
+    def _get_transition_handlers(self):
+        return OrderedDict([
+            (('not_circulate', 'circulate'), lambda: self.circulate_commanded),
+            (('circulate', 'not_circulate'), lambda: not self.circulate_commanded),
+        ])
+
+    def get_bath_temperature(self):
+        """
+        Gets the external temperature of the bath.
+
+        :return: The external temperature.
+        """
+        return self.temperature
+
+    def get_external_temperature(self):
+        """
+        Gets the temperature of the bath.
+
+        :return: The current bath temperature.
+        """
+        return self.external_temperature
+
+    def get_power(self):
+        """
+        Gets the heating power currently being used.
+
+        :return: The heating power.
+        """
+        return self.heating_power
+
+    def get_set_point(self):
+        """
+        Gets the set point the user requested.
+
+        :return: The set point temperature.
+        """
+        return self.set_point_temperature
+
+    def get_high_limit(self):
+        """
+        Gets the high limit set for the bath.
+
+        These are usually set manually in the hardware.
+
+        :return: The high limit.
+        """
+        return self.temperature_high_limit
+
+    def get_low_limit(self):
+        """
+        Gets the low limit set for the bath.
+
+        These are usually set manually in the hardware.
+
+        :return: The low limit.
+        """
+        return self.temperature_low_limit
+
+    def get_circulating(self):
+        """
+        Gets whether the bath is circulating.
+
+        This means the heater is on?
+
+        :return: O for off, 1 for on.
+        """
+        return self.is_circulating
+
+    def get_version(self):
+        """
+        Gets the Julabo version number.
+
+        :return: Version string.
+        """
+        return "JULABO FP50_MH Simulator, ISIS"
+
+    def get_status(self):
+        """
+        Not sure what a real device returns as the manual is a bit vague.
+        It will return error codes but it is not clear what it returns if everything is okay.
+
+        :return: String
+        """
+        return "Hello from the simulated Julabo"
+
+    def set_set_point(self, param):
```

Comment:
This would be an example for `check_limits` (very new), which would simplify the function to:

```
@check_limits('temperature_low_limit', 'temperature_high_limit', silent=True)
def set_set_point(self, new_setpoint):
    self.set_point_temperature = new_setpoint
    return ""
```

The `return ""` part could even be achieved by having a `return_mapping` on the device's interface. Would it help to make it possible to have a default return mapping on the interface? Then it would only need to be overridden for cases where it's different. Here it's probably something like:

```
def default_return_mapping(value):
    return "" if value is None else value
```